Reaper doesn't appear to be thread safe

**Describe the bug**
Having recently upgraded from 0.4.0 to 0.5.1 I noticed some tests started failing with:
```
could not start container: failed to create container: connecting to reaper failed: Connecting to Ryuk on  failed: dial tcp: missing address
```
We run a number of tests in parallel (in the same package) spinning up a few containers (less than 5). If we remove `t.Parallel()` the test pass successfully, but they take longer to complete.

**To Reproduce**
Can you write a unit test to reproduce your issue?
```

import (
	"context"
	"fmt"
	"github.com/testcontainers/testcontainers-go"
	"github.com/testcontainers/testcontainers-go/wait"
	"testing"
	"time"
)

func TestParallelContainers(t *testing.T) {
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
	t.Cleanup(cancel)

	for i := 0; i < 3; i++ {
		t.Run(fmt.Sprintf("iteration_%d", i), func(t *testing.T) {
			t.Parallel()
			createTestContainer(t, ctx)
		})
	}
}

func createTestContainer(t *testing.T, ctx context.Context) int {
	req := testcontainers.ContainerRequest{
		Image:        "localstack/localstack:latest",
		ExposedPorts: []string{"4584/tcp", "8080/tcp"},
		Env: map[string]string{
			"SERVICES": "secretsmanager",
		},
		WaitingFor: wait.ForListeningPort("4584/tcp"),
		//SkipReaper: true,
	}
	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
		ContainerRequest: req,
		Started:          true,
	})
	if err != nil {
		t.Fatalf("could not start container: %v", err)
	}
	port, err := container.MappedPort(ctx, "4584")
	if err != nil {
		t.Fatalf("could not get mapped port: %v", err)
	}

	t.Cleanup(func() {
		container.Terminate(context.Background())
	})

	return port.Int()
}

```

**Expected behavior**
It should be possible to start containers in parallel.

**Additional context**
Test output:
```
=== RUN   TestParallelContainers
--- FAIL: TestParallelContainers (0.00s)
=== RUN   TestParallelContainers/iteration_0
=== PAUSE TestParallelContainers/iteration_0
=== CONT  TestParallelContainers/iteration_0
    --- FAIL: TestParallelContainers/iteration_0 (0.01s)
=== RUN   TestParallelContainers/iteration_1
=== PAUSE TestParallelContainers/iteration_1
=== CONT  TestParallelContainers/iteration_1
    TestParallelContainers/iteration_1: demo_test.go:39: could not start container: failed to create container: connecting to reaper failed: Connecting to Ryuk on  failed: dial tcp: missing address
    TestParallelContainers/iteration_0: demo_test.go:39: could not start container: failed to create container: connecting to reaper failed: Connecting to Ryuk on  failed: dial tcp: missing address
2020/05/12 09:57:27 Starting container id: 87b3a8c0ba74 image: quay.io/testcontainers/ryuk:0.2.3
2020/05/12 09:57:28 Waiting for container id 87b3a8c0ba74 image: quay.io/testcontainers/ryuk:0.2.3
2020/05/12 09:57:28 Container is ready id: 87b3a8c0ba74 image: quay.io/testcontainers/ryuk:0.2.3
2020/05/12 09:57:28 Starting container id: c3bf40970170 image: localstack/localstack:latest
2020/05/12 09:57:28 Waiting for container id c3bf40970170 image: localstack/localstack:latest
2020/05/12 09:57:43 Container is ready id: c3bf40970170 image: localstack/localstack:latest
    --- FAIL: TestParallelContainers/iteration_1 (0.01s)
=== RUN   TestParallelContainers/iteration_2
=== PAUSE TestParallelContainers/iteration_2
=== CONT  TestParallelContainers/iteration_2
    --- PASS: TestParallelContainers/iteration_2 (16.15s)
FAIL

Process finished with exit code 1
```

--- a/container_test.go
+++ b/container_test.go
@@ -4,6 +4,7 @@ import (
 	"archive/tar"
 	"bytes"
 	"context"
+	"fmt"
 	"io"
 	"io/ioutil"
 	"strings"
@@ -302,4 +303,44 @@ func Test_GetLogsFromFailedContainer(t *testing.T) {
 	if strings.Contains(log, "I was not expecting this") == false {
 		t.Fatalf("could not find expected log in %s", log)
 	}
+}
+
+func TestShouldStartContainersInParallel(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	t.Cleanup(cancel)
+
+	for i := 0; i < 3; i++ {
+		t.Run(fmt.Sprintf("iteration_%d", i), func(t *testing.T) {
+			t.Parallel()
+			createTestContainer(t, ctx)
+		})
+	}
+}
+
+func createTestContainer(t *testing.T, ctx context.Context) int {
+	req := ContainerRequest{
+		Image:        "localstack/localstack:latest",
+		ExposedPorts: []string{"4584/tcp", "8080/tcp"},
+		Env: map[string]string{
+			"SERVICES": "secretsmanager",
+		},
+		WaitingFor: wait.ForListeningPort("4584/tcp"),
+	}
+	container, err := GenericContainer(ctx, GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	if err != nil {
+		t.Fatalf("could not start container: %v", err)
+	}
+	port, err := container.MappedPort(ctx, "4584")
+	if err != nil {
+		t.Fatalf("could not get mapped port: %v", err)
+	}
+
+	t.Cleanup(func() {
+		container.Terminate(context.Background())
+	})
+
+	return port.Int()
 }

--- a/reaper.go
+++ b/reaper.go
@@ -8,6 +8,7 @@ import (
 	"github.com/testcontainers/testcontainers-go/wait"
 	"net"
 	"strings"
+	"sync"
 	"time"
 
 	"github.com/pkg/errors"
@@ -22,6 +23,7 @@ const (
 )
 
 var reaper *Reaper // We would like to create reaper only once
+var mutex sync.Mutex
 
 // ReaperProvider represents a provider for the reaper to run itself with
 // The ContainerProvider interface should usually satisfy this as well, so it is pluggable
@@ -38,6 +40,8 @@ type Reaper struct {
 
 // NewReaper creates a Reaper with a sessionID to identify containers and a provider to use
 func NewReaper(ctx context.Context, sessionID string, provider ReaperProvider, reaperImageName string) (*Reaper, error) {
+	mutex.Lock()
+	defer mutex.Unlock()
 	// If reaper already exists re-use it
 	if reaper != nil {
 		return reaper, nil

